core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory.java

```diff
@@ -172,54 +175,90 @@ public void handleResolvedAddressGroups(
         subchannels.put(addressGroup, subchannel);
         subchannel.requestConnection();
       }
+      
+      // Note that the active list is not changed by subchannel additions,
+      // since they are added in IDLE (non-READY) state.
+      boolean activeListChanged = false;
 
       // Shutdown subchannels for removed addresses.
       for (EquivalentAddressGroup addressGroup : removedAddrs) {
         Subchannel subchannel = subchannels.remove(addressGroup);
-        subchannel.shutdown();
+        // Active list only changes if we remove a subchannel in READY state
+        activeListChanged = activeListChanged || isReady(subchannel);
+        shutdownSubchannel(subchannel);
       }
 
-      updateBalancingState(getAggregatedState(), getAggregatedError());
+      updateBalancingState(activeListChanged || stickinessStateChanged);
```

Comment:
While the logic is clear enough as it is, I still don't like the little addition of complexity. The list change handling is not in the RPC critical path, and the overhead of creating and discarding a new picker doesn't seem to be significant enough to justify the added code complexity, e.g., the boolean argument here passes ad-hoc information across function boundaries just because your logic has to be divided into multiple functions instead of being one -- not saying it's wrong, but I'd like to avoid such situation if possible.

Reply:
Sure, I will rework to always re-generate and compare the pickers.